nomad/deploymentwatcher/deployment_watcher.go

```diff
@@ -371,7 +371,13 @@ func (w *deploymentWatcher) watch() {
 				// Description should include that the job is being rolled back to
 				// version N
 				if j != nil {
-					desc = structs.DeploymentStatusDescriptionRollback(desc, j.Version)
+					// only revert if job being changed has a different spec
```

Comment:
1. Capitalize the start of your comments.
2. Can we add more detail in the comment explaining how this can avoid an infinite revert cycle.